src/main/scala/logger/SymbExLogger.scala

```diff
@@ -192,36 +185,42 @@ import scala.util.{Failure, Success, Try}
   * // SymbExLogger.currentLog().markReachable(uidBranchPoint)
   */
 
-object SymbExLogger {
-  /** Collection of logged Method/Predicates/Functions. **/
-  var memberList: Seq[SymbLog] = Seq[SymbLog]()
-  private var uidCounter = 0
-
-  var enabled = false
-  var logConfig: LogConfig = LogConfig.default()
-
-  def freshUid(): Int = {
-    val uid = uidCounter
-    uidCounter = uidCounter + 1
-    uid
+case object SymbExLogger {
+  def ofConfig(c: Config): SymbExLogger[_ <: MemberSymbExLogger] = {
```

Comment:
It took me a while to realise that `c` is Silicon's config, and `config` further down (and in other files, e.g. `DataRecord.scala` ) is the symbex-logger config. Since `config` everywhere else in Silicon denotes Silicon's configuration object, it would be best to not overload the name. If it's not too much work, please rename the symbex-logger config object to, e.g. `symbExConfig`  or `loggerConfig`.